arm9/src/poke_overlay.c

```diff
@@ -0,0 +1,203 @@
+#include "global.h"
+#include "nitro.h"
+#include "FS_file.h"
+#include "FS_overlay.h"
+
+#define OVERLAY_LOAD_WRAM 0
+#define OVERLAY_LOAD_ITCM 1
+#define OVERLAY_LOAD_DTCM 2
+
+extern void ErrorHandling(void);
+
+struct LoadedOverlay {
+    FSOverlayID id;
+    BOOL active;
+};
+
+extern struct LoadedOverlay gLoadedOverlays[3][8];
+
+struct LoadedOverlay* GetLoadedOverlaysInRegion(int);
+BOOL GetOverlayRamBounds(FSOverlayID, void**, void**);
+BOOL CanOverlayBeLoaded(FSOverlayID);
+int GetOverlayLoadDestination(FSOverlayID);
+BOOL LoadOverlayNormal(MIProcessor, FSOverlayID);
+BOOL LoadOverlayNoInit(MIProcessor, FSOverlayID);
+BOOL LoadOverlayNoInitAsync(MIProcessor, FSOverlayID);
+
+THUMB_FUNC void FreeOverlayAllocation(struct LoadedOverlay * loaded)
+{
+    if (loaded->active != TRUE)
+        ErrorHandling();
+    if (FS_UnloadOverlay(MI_PROCESSOR_ARM9, loaded->id) != TRUE)
+        ErrorHandling();
+    loaded->active = FALSE;
+}
+
+THUMB_FUNC void UnloadOverlayByID(FSOverlayID id)
+{
+    struct LoadedOverlay * loaded = GetLoadedOverlaysInRegion(GetOverlayLoadDestination(id));
+    int i;
+    for (i = 0; i < 8; i++)
+    {
+        if (loaded[i].active == TRUE && loaded[i].id == id)
+        {
+            FreeOverlayAllocation(&loaded[i]);
+            break;
+        }
+    }
+}
+
+THUMB_FUNC int GetOverlayLoadDestination(FSOverlayID id)
+{
+    FSOverlayInfo info;
+    u8 *end;
+    u8 *start;
+    if (FS_LoadOverlayInfo(&info, MI_PROCESSOR_ARM9, id) != TRUE)
+        ErrorHandling();
+    start = (u8 *)HW_ITCM_IMAGE;
+    end = (u8 *)HW_ITCM_END;
+    if (info.header.ram_address <= end && info.header.ram_address >= start)
+        return OVERLAY_LOAD_ITCM;
+    start = (u8 *)SDK_AUTOLOAD_DTCM_START;
+    end = start + HW_DTCM_SIZE;
+    if (info.header.ram_address <= end && info.header.ram_address >= start)
+        return OVERLAY_LOAD_DTCM;
+    return OVERLAY_LOAD_WRAM;
+}
+
+THUMB_FUNC BOOL HandleLoadOverlay(FSOverlayID id, int a1)
+{
+    u32 sp0 = FS_DMA_NOT_USE;
+    struct LoadedOverlay *r3;
+    int r7;
```

Comment:
int vs s32?

Reply:
on this platform, `int` and `s32` are one and the same

Reply:
It would be best to use the specified types anyways, wouldn't it? since we use the SDK definitions everywhere else as well, it's good for consistency.